`unnecessary-ellipsis` checker: `AstroidError` with list in dict (Could not find list in dict's children)

### Bug description

```python
# pylint: disable=missing-module-docstring
x = {'a': [...]}
```

leads to an `AstroidError`.

I bisected this to 78eed6af303e0e9854e43528d17d2ada96f11c09, part of #5470 ("New checker - Detect use of unnecessary ellipsis").

### Configuration

_No response_

### Command used

```shell
pylint repro3.py
```


### Pylint output

```shell
Exception on node <Const.Ellipsis l.2 at 0x7fb5c8bd45e0> in file '/home/florian/proj/pylint/repro3.py'
Traceback (most recent call last):
  File "/home/florian/proj/pylint/pylint/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File "/home/florian/proj/pylint/pylint/pylint/checkers/ellipsis_checker.py", line 38, in visit_const
    len(node.parent.parent.child_sequence(node.parent)) > 1
  File "/home/florian/proj/pylint/astroid/astroid/nodes/node_ng.py", line 397, in child_sequence
    raise AstroidError(msg % (repr(child), repr(self)))
astroid.exceptions.AstroidError: Could not find <List.list l.2 at 0x7fb5c8bd4580> in <Dict.dict l.2 at 0x7fb5c8bd77c0>'s children
************* Module repro3
repro3.py:1:0: F0002: repro3.py: Fatal error while checking 'repro3.py'. Please open an issue in our bug tracker so we address this. [...]
```


### Expected behavior

No crash, and probably no error either.

### Pylint version

```shell
pylint 2.13.2
astroid 2.11.2
Python 3.8.12 (default, Mar  3 2022, 18:19:28) 
[GCC 11.2.0]
```

(environment in which I originally encountered the bug - output above is from a git install of both astroid and pylint)

### OS / Environment

Archlinux

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -28,6 +28,10 @@ Release date: TBA
 
 * Fix bug where specifically enabling just ``await-outside-async`` was not possible.
 
+* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used inside of a list.
+
+  Closes #6037
+
 ..
   Insert your changelog randomly, it will reduce merge conflicts
   (Ie. not necessarily at the end)

--- a/ChangeLog
+++ b/ChangeLog
@@ -45,6 +45,7 @@ Release date: TBA
 inside of a container.
 
   Closes #6037
+  Closes #6048
 
 What's New in Pylint 2.13.3?
 ============================

--- a/ChangeLog
+++ b/ChangeLog
@@ -41,8 +41,7 @@ Release date: TBA
 
   Closes #6028
 
-* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used
-inside of a container.
+* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used inside of a container.
 
   Closes #6037
   Closes #6048

--- a/ChangeLog
+++ b/ChangeLog
@@ -46,7 +46,6 @@ inside of a container.
 
   Closes #6037
 
-
 What's New in Pylint 2.13.3?
 ============================
 Release date: 2022-03-29

--- a/ChangeLog
+++ b/ChangeLog
@@ -41,7 +41,8 @@ Release date: TBA
 
   Closes #6028
 
-* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used inside of a list.
+* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used
+inside of a container.
 
   Closes #6037
 

--- a/ChangeLog
+++ b/ChangeLog
@@ -28,10 +28,6 @@ Release date: TBA
 
 * Fix bug where specifically enabling just ``await-outside-async`` was not possible.
 
-* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used inside of a list.
-
-  Closes #6037
-
 ..
   Insert your changelog randomly, it will reduce merge conflicts
   (Ie. not necessarily at the end)
@@ -44,6 +40,10 @@ Release date: TBA
 * Include ``testing_pylintrc`` in source and wheel distributions.
 
   Closes #6028
+
+* Fix crash for ``unneccessary-ellipsis`` checker when an ellipsis is used inside of a list.
+
+  Closes #6037
 
 
 What's New in Pylint 2.13.3?

--- a/pylint/checkers/ellipsis_checker.py
+++ b/pylint/checkers/ellipsis_checker.py
@@ -39,16 +39,17 @@ class EllipsisChecker(BaseChecker):
            For example: A function consisting of an ellipsis followed by a
            return statement on the next line.
         """
+        #breakpoint()
         if (
             node.pytype() == "builtins.Ellipsis"
             and not isinstance(
                 node.parent,
                 (
-                    nodes.Assign,
                     nodes.AnnAssign,
-                    nodes.Call,
                     nodes.Arguments,
-                    nodes.List,
+                    nodes.Assign,
+                    nodes.BaseContainer,
+                    nodes.Call,
                 ),
             )
             and (

--- a/pylint/checkers/ellipsis_checker.py
+++ b/pylint/checkers/ellipsis_checker.py
@@ -39,7 +39,6 @@ class EllipsisChecker(BaseChecker):
            For example: A function consisting of an ellipsis followed by a
            return statement on the next line.
         """
-        # breakpoint()
         if (
             node.pytype() == "builtins.Ellipsis"
             and not isinstance(

--- a/pylint/checkers/ellipsis_checker.py
+++ b/pylint/checkers/ellipsis_checker.py
@@ -43,7 +43,13 @@ class EllipsisChecker(BaseChecker):
             node.pytype() == "builtins.Ellipsis"
             and not isinstance(
                 node.parent,
-                (nodes.Assign, nodes.AnnAssign, nodes.Call, nodes.Arguments, nodes.List),
+                (
+                    nodes.Assign,
+                    nodes.AnnAssign,
+                    nodes.Call,
+                    nodes.Arguments,
+                    nodes.List,
+                ),
             )
             and (
                 len(node.parent.parent.child_sequence(node.parent)) > 1

--- a/pylint/checkers/ellipsis_checker.py
+++ b/pylint/checkers/ellipsis_checker.py
@@ -39,7 +39,7 @@ class EllipsisChecker(BaseChecker):
            For example: A function consisting of an ellipsis followed by a
            return statement on the next line.
         """
-        #breakpoint()
+        # breakpoint()
         if (
             node.pytype() == "builtins.Ellipsis"
             and not isinstance(

--- a/pylint/checkers/ellipsis_checker.py
+++ b/pylint/checkers/ellipsis_checker.py
@@ -43,7 +43,7 @@ class EllipsisChecker(BaseChecker):
             node.pytype() == "builtins.Ellipsis"
             and not isinstance(
                 node.parent,
-                (nodes.Assign, nodes.AnnAssign, nodes.Call, nodes.Arguments),
+                (nodes.Assign, nodes.AnnAssign, nodes.Call, nodes.Arguments, nodes.List),
             )
             and (
                 len(node.parent.parent.child_sequence(node.parent)) > 1

--- a/tests/functional/u/unnecessary/unnecessary_ellipsis.py
+++ b/tests/functional/u/unnecessary/unnecessary_ellipsis.py
@@ -103,6 +103,12 @@ def func_with_ellipsis_default_arg(a = ...) -> None:
     "Some docstring."
 
 
-# Ignore if the ellipsis is inside a list
-x = [...]
-y = {'a': [...]}
+# Ignore if the ellipsis is inside a container:
+my_list = [...]
+my_tuple = (...,)
+my_set = {...}
+
+# Ellipsis inside a container which is a value in a dictionary
+a = {'x': [...]}
+b = {'x': {...}}
+c = {'x': (...,)}

--- a/tests/functional/u/unnecessary/unnecessary_ellipsis.py
+++ b/tests/functional/u/unnecessary/unnecessary_ellipsis.py
@@ -101,3 +101,8 @@ class MyIntegerList(List[int]):
 # Ellipsis is allowed as a default argument
 def func_with_ellipsis_default_arg(a = ...) -> None:
     "Some docstring."
+
+
+# Ignore if the ellipsis is inside a list
+x = [...]
+y = {'a': [...]}

--- a/tests/functional/u/unnecessary/unnecessary_ellipsis.py
+++ b/tests/functional/u/unnecessary/unnecessary_ellipsis.py
@@ -109,6 +109,6 @@ my_tuple = (...,)
 my_set = {...}
 
 # Ellipsis inside a container which is a value in a dictionary
-a = {'x': [...]}
-b = {'x': {...}}
-c = {'x': (...,)}
+mydict1 = {'x': [...]}
+mydict2 = {'x': {...}}
+mydict3 = {'x': (...,)}

